api/groups/eventsData.go

```diff
@@ -1,20 +1,52 @@
 package groups
 
 import (
-	"encoding/json"
+	"fmt"
 
 	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
 	"github.com/multiversx/mx-chain-core-go/data/block"
 	"github.com/multiversx/mx-chain-core-go/data/outport"
 	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-notifier-go/common"
 	"github.com/multiversx/mx-chain-notifier-go/data"
 )
 
-// UnmarshallBlockDataV1 will try to unmarshal block data with old format
-func UnmarshallBlockDataV1(marshalledData []byte) (*data.SaveBlockData, error) {
+type eventsDataHandler struct {
+	marshaller         marshal.Marshalizer
+	internalMarshaller marshal.Marshalizer
+	emptyBlockCreator  EmptyBlockCreatorContainer
+}
+
+// NewEventsDataHandler will create a new data handler component
+func NewEventsDataHandler(marshaller marshal.Marshalizer, internalMarshaller marshal.Marshalizer) (*eventsDataHandler, error) {
+	if check.IfNil(marshaller) {
+		return nil, common.ErrNilMarshaller
+	}
+	if check.IfNil(internalMarshaller) {
+		return nil, fmt.Errorf("%w (internal)", common.ErrNilMarshaller)
+	}
+
+	edh := &eventsDataHandler{
+		marshaller:         marshaller,
+		internalMarshaller: internalMarshaller,
+	}
+
+	emptyBlockContainer, err := createEmptyBlockCreatorContainer()
+	if err != nil {
+		return nil, err
+	}
+
+	edh.emptyBlockCreator = emptyBlockContainer
+
+	return edh, nil
+}
+
+// UnmarshallBlockDataOld will try to unmarshal block data with old format
+func (edh *eventsDataHandler) UnmarshallBlockDataOld(marshalledData []byte) (*data.SaveBlockData, error) {
```

Comment:
For another PR: in order to avoid this kind of duplicated methods (ending with old) that will almost require a switch somewhere (either call the new method or the old method), we can have another component for this "old" mechanism and a factory/proxy that will switch the implementation at runtime.